Test with resize fails on firefox:headless

```javascript
import { ClientFunction } from 'testcafe';

fixture `My fixture`
    .page `https://testcafe.devexpress.com/Example/`;

const getWindowProperty = new ClientFunction(propName => window[propName]);

test('test', async t => {
    await t
        .resizeWindow(600, 400)
        .expect(getWindowProperty('innerWidth')).eql(600)
        .expect(getWindowProperty('innerHeight')).eql(400);
});
```
 AssertionError: expected 617 to deeply equal 600

 Browser: Firefox 56.0.0 / Windows 8.1 0.0.0

--- a/src/browser/provider/built-in/chrome/index.js
+++ b/src/browser/provider/built-in/chrome/index.js
@@ -3,18 +3,7 @@ import getRuntimeInfo from './runtime-info';
 import getConfig from './config';
 import { start as startLocalChrome, stop as stopLocalChrome } from './local-chrome';
 import * as cdp from './cdp';
-
-
-/*eslint-disable no-undef*/
-function getWindowDimensionsInfo () {
-    return {
-        width:  window.innerWidth,
-        height: window.innerHeight
-    };
-}
-/*eslint-enable no-undef*/
-
-const GET_WINDOW_DIMENSIONS_INFO_SCRIPT = getWindowDimensionsInfo.toString();
+import { GET_WINDOW_DIMENSIONS_INFO_SCRIPT } from '../../utils/client-functions';
 
 
 export default {

--- a/src/browser/provider/built-in/firefox/marionette-client.js
+++ b/src/browser/provider/built-in/firefox/marionette-client.js
@@ -4,14 +4,13 @@ import promisifyEvent from 'promisify-event';
 import EventEmitter from 'events';
 import { writeFile } from '../../../../utils/promisified-functions';
 import delay from '../../../../utils/delay';
+import { GET_WINDOW_DIMENSIONS_INFO_SCRIPT } from '../../utils/client-functions';
 
 
 const CONNECTION_READY_TIMEOUT   = 300;
 const MAX_CONNECTION_RETRY_COUNT = 10;
-const MAX_RESIZE_RETRY_COUNT     = 5;
+const MAX_RESIZE_RETRY_COUNT     = 2;
 const HEADER_SEPARATOR           = ':';
-
-const GET_PAGE_SIZE_SCRIPT = 'return { width: window.innerWidth, height: window.innerHeight }';
 
 module.exports = class MarionetteClient {
     constructor (port = 2828, host = '127.0.0.1') {
@@ -159,8 +158,8 @@ module.exports = class MarionetteClient {
         this.buffer = null;
     }
 
-    async executeScript (expression) {
-        return await this._getResponse({ command: 'executeScript', parameters: { script: expression } });
+    async executeScript (code) {
+        return await this._getResponse({ command: 'executeScript', parameters: { script: `return (${code})()` } });
     }
 
     async takeScreenshot (path) {
@@ -170,8 +169,8 @@ module.exports = class MarionetteClient {
     }
 
     async setWindowSize (width, height) {
-        var { value: pageRect } = await this.executeScript(GET_PAGE_SIZE_SCRIPT);
-        var attemptCounter       = 0;
+        var { value: pageRect } = await this.executeScript(GET_WINDOW_DIMENSIONS_INFO_SCRIPT);
+        var attemptCounter      = 0;
 
         while (attemptCounter++ < MAX_RESIZE_RETRY_COUNT && (pageRect.width !== width || pageRect.height !== height)) {
             var currentRect = await this._getResponse({ command: 'getWindowRect' });
@@ -187,7 +186,7 @@ module.exports = class MarionetteClient {
                 }
             });
 
-            ({ value: pageRect } = await this.executeScript(GET_PAGE_SIZE_SCRIPT));
+            ({ value: pageRect } = await this.executeScript(GET_WINDOW_DIMENSIONS_INFO_SCRIPT));
         }
     }
 

--- a/src/browser/provider/built-in/firefox/marionette-client.js
+++ b/src/browser/provider/built-in/firefox/marionette-client.js
@@ -8,9 +8,10 @@ import delay from '../../../../utils/delay';
 
 const CONNECTION_READY_TIMEOUT   = 300;
 const MAX_CONNECTION_RETRY_COUNT = 10;
+const MAX_RESIZE_RETRY_COUNT     = 5;
 const HEADER_SEPARATOR           = ':';
 
-const GET_PAGE_SIZE_SCRIPT = 'return {width:document.documentElement.clientWidth, height: document.documentElement.clientHeight}';
+const GET_PAGE_SIZE_SCRIPT = 'return { width: window.innerWidth, height: window.innerHeight }';
 
 module.exports = class MarionetteClient {
     constructor (port = 2828, host = '127.0.0.1') {
@@ -170,8 +171,9 @@ module.exports = class MarionetteClient {
 
     async setWindowSize (width, height) {
         var { value: pageRect } = await this.executeScript(GET_PAGE_SIZE_SCRIPT);
+        var attemptCounter       = 0;
 
-        while (pageRect.width !== width || pageRect.height !== height) {
+        while (attemptCounter++ < MAX_RESIZE_RETRY_COUNT && (pageRect.width !== width || pageRect.height !== height)) {
             var currentRect = await this._getResponse({ command: 'getWindowRect' });
 
             await this._getResponse({

--- a/src/browser/provider/built-in/firefox/runtime-info.js
+++ b/src/browser/provider/built-in/firefox/runtime-info.js
@@ -24,7 +24,14 @@ async function generatePrefs (profileDir, port) {
         'user_pref("browser.startup.homepage_override.mstone","ignore");',
         'user_pref("browser.tabs.warnOnCloseOtherTabs", false);',
         'user_pref("browser.tabs.warnOnClose", false);',
-        'user_pref("toolkit.telemetry.reportingpolicy.firstRun", false);'
+        'user_pref("toolkit.telemetry.reportingpolicy.firstRun", false);',
+        'user_pref("toolkit.telemetry.enabled", false);',
+        'user_pref("toolkit.telemetry.rejected", true);',
+        'user_pref("datareporting.healthreport.uploadEnabled", false);',
+        'user_pref("datareporting.healthreport.service.enabled", false);',
+        'user_pref("datareporting.healthreport.service.firstRun", false);',
+        'user_pref("datareporting.policy.dataSubmissionEnabled", false);',
+        'user_pref("datareporting.policy.dataSubmissionPolicyBypassNotification", true);'
     ];
 
     if (port) {

--- a/src/browser/provider/index.js
+++ b/src/browser/provider/index.js
@@ -2,27 +2,8 @@ import Promise from 'pinkie';
 import browserTools from 'testcafe-browser-tools';
 import OS from 'os-family';
 import delay from '../../utils/delay';
+import { GET_TITLE_SCRIPT, GET_WINDOW_DIMENSIONS_INFO_SCRIPT } from './utils/client-functions';
 
-
-/*eslint-disable no-undef*/
-function getTitle () {
-    return document.title;
-}
-
-function getWindowDimensionsInfo () {
-    return {
-        width:           window.innerWidth,
-        height:          window.innerHeight,
-        outerWidth:      window.outerWidth,
-        outerHeight:     window.outerHeight,
-        availableWidth:  screen.availWidth,
-        availableHeight: screen.availHeight
-    };
-}
-/*eslint-disable no-undef*/
-
-const GET_TITLE_SCRIPT                  = getTitle.toString();
-const GET_WINDOW_DIMENSIONS_INFO_SCRIPT = getWindowDimensionsInfo.toString();
 
 const BROWSER_OPENING_DELAY = 2000;
 

--- a/src/browser/provider/utils/client-functions.js
+++ b/src/browser/provider/utils/client-functions.js
@@ -1,0 +1,19 @@
+/*eslint-disable no-undef*/
+function getTitle () {
+    return document.title;
+}
+
+function getWindowDimensionsInfo () {
+    return {
+        width:           window.innerWidth,
+        height:          window.innerHeight,
+        outerWidth:      window.outerWidth,
+        outerHeight:     window.outerHeight,
+        availableWidth:  screen.availWidth,
+        availableHeight: screen.availHeight
+    };
+}
+/*eslint-disable no-undef*/
+
+export const GET_TITLE_SCRIPT                  = getTitle.toString();
+export const GET_WINDOW_DIMENSIONS_INFO_SCRIPT = getWindowDimensionsInfo.toString();

